src/platform/plugins/shared/content_management/server/routes/list/query_builder.ts

```diff
@@ -0,0 +1,328 @@
+ * attributes in the document source. Optionally creates an additional runtime field for
+ * custom sort fields (e.g., `sortable_myField`).
+ *
+ * The runtime mappings use Painless scripts to access attributes stored at `source[type]`
+ * (not `source[type].attributes`), matching Kibana's saved object document structure.
+ *
+ * @param sortField - Optional custom field name to create a runtime mapping for.
+ *   Root fields (`updatedAt`, `createdAt`, `updatedBy`, `createdBy`, `managed`) and
+ *   `title` are handled specially and don't need additional runtime mappings.
+ * @returns Elasticsearch runtime field mappings.
+ *
+ * @example
+ * // For sorting by title:
+ * buildRuntimeMappings(); // Returns { sortable_title: {...} }
+ *
+ * @example
+ * // For sorting by a custom attribute:
+ * buildRuntimeMappings('version'); // Returns { sortable_title: {...}, sortable_version: {...} }
+ */
+export const buildRuntimeMappings = (sortField?: string): estypes.MappingRuntimeFields => {
+  // Known root fields that don't need runtime mappings for sorting.
+  const rootFields = ['updatedAt', 'createdAt', 'updatedBy', 'createdBy', 'managed'];
+
+  const mappings: estypes.MappingRuntimeFields = {
+    sortable_title: {
+      type: 'keyword',
+      script: {
+        source: `
+          String docType = params._source.type;
+          if (docType != null &&
+              params._source.containsKey(docType) &&
+              params._source[docType].containsKey('title')) {
+            emit(params._source[docType].title.toLowerCase());
+          } else {
+            emit('');
+          }
+        `,
+      },
+    },
+  };
+
+  // Add dynamic runtime mapping for custom attribute sort fields.
+  // Uses script params to avoid script injection vulnerabilities.
+  if (sortField && !rootFields.includes(sortField) && sortField !== 'title') {
+    mappings[`sortable_${sortField}`] = {
+      type: 'keyword',
+      script: {
+        source: `
+          String docType = params._source.type;
+          String fieldName = params.fieldName;
+          if (docType != null &&
+              params._source.containsKey(docType) &&
+              params._source[docType].containsKey(fieldName)) {
+            def val = params._source[docType][fieldName];
+            emit(val != null ? val.toString().toLowerCase() : '');
+          } else {
+            emit('');
+          }
+        `,
+        params: {
+          fieldName: sortField,
+        },
+      },
+    };
+  }
+
+  return mappings;
+};
+
+/**
+ * Maps sort field names from the request to the actual ES field paths.
+ */
+const mapSortField = (field: string): string => {
+  // For title, use the runtime mapping.
+  if (field === 'title') {
+    return 'sortable_title';
+  }
+
+  // Map root fields to snake_case.
+  const rootFieldMap: Record<string, string> = {
+    updatedAt: 'updated_at',
+    createdAt: 'created_at',
+    updatedBy: 'updated_by',
```

Comment:
I remembered why created_by / updated_by weren't sortable in the UI. this is because we store user id and it doesn't make sense to sort by user id, we should be sorting by user-facing naming. 

Is this handled somehow? Otherwise we should keep this unsortable

Reply:
Thanks for that insight!  I can adjust, since I'm gathering the metadata for the response!